libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Instrumentation/SFMethodInterceptor.m

```diff
@@ -194,7 +208,9 @@ - (void)swizzle {
         Method originalMethod = class_getInstanceMethod(self.classToIntercept, self.selectorToIntercept);
         const char *methodTypes = method_getTypeEncoding(originalMethod);
         
-        InterceptorsForClassAndSelector[interceptorKey(self.classToIntercept, self.selectorToIntercept)] = self;
+        @synchronized (InterceptorsForClassAndSelector) {
+            InterceptorsForClassAndSelector[interceptorKey(self.classToIntercept, self.selectorToIntercept)] = self;
```

Comment:
NB: There seem like some unhandled edge cases with `interceptorKey`, given that it's not technically unique: you can have a class an instance method with the same signature, which is going to equate to the same interceptor object here.

Unfortunately, since `forwardingTargetForSelector:` doesn't provide information on whether the selector is a class or instance selector, we don't have a great way to reverse-engineer that information at swizzle time.  Not sure how to handle that case at this point.  FRI @jeanbovet .